server/channels/api4/system.go

```diff
@@ -596,6 +596,16 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	}()
 
 	location = res.Header.Get("Location")
+
+	// If the location length is > 2100, we can probably ignore. Fixes https://mattermost.atlassian.net/browse/MM-54219
+	if len(location) > 2100 {
+		// Treating as a "failure". Cache failures to prevent retries.
+		redirectLocationDataCache.SetWithExpiry(url, "", 1*time.Hour)
+		// Always return a success status and a JSON string to limit information returned to client.
+		w.Write([]byte(model.MapToJSON(m)))
```

Comment:
Basically I was following the comments, to limit information returned to the client. I don't have enough context to decide whether or not we should return a 400 here (because if we do here, why not above too?).

Reply:
Alright, it shouldn't be a big deal. It's a breaking change regardless since there's info loss.